nixos/modules/programs/mosh.nix

```diff
@@ -16,10 +16,28 @@ in
       default = false;
       type = lib.types.bool;
     };
+    withUtempter = mkOption {
+      description = ''
+        Whether to enable libutempter for mosh.
+        This is required so that mosh can write to /var/run/utmp (which can be queried with `who` to display currently connected user sessions).
+        Note, this will add a suid wrapper!
+      '';
+      default = true;
+      type = lib.types.bool;
+    };
   };
 
   config = mkIf cfg.enable {
-    environment.systemPackages = with pkgs; [ mosh ];
+    environment.systemPackages = [ ( pkgs.mosh.override { inherit (cfg) withUtempter; } ) ];
```

Comment:
This will trigger a local rebuild of `mosh` if `withUtempter=true;` because Hydra only builds the default version.

You could add `mosh-with-utempter = mosh.override { withUtempter = true; }` to `pkgs/top-level/all-packages.nix` to have Hydra build that version too. 

Or you can build mosh with utempter support by default, provided it doesn't break when the setuid wrapper is missing.

Reply:
Yup, I feared as much. Will change that, thanks :)

Reply:
> In any case: compiling against utempter without having the wrapper in place has no negative consequences. /var/run/utmp will not be written, but mosh continues working.

Then I believe it is better to build `mosh` with `utempter` support by default, so we don't have to maintain two variants. It shouldn't add much to closure size, and simplifies the module.

Reply:
fine, just ping me when it's ready.